builtin/plugins/chainconfig/chainconfig.go

```diff
@@ -234,12 +235,36 @@ func (c *ChainConfig) ListFeatures(ctx contract.StaticContext, req *ListFeatures
 		}
 		features = append(features, feature)
 	}
-
+	featuresFromState := ctx.GetEnabledFeatures()
```

Comment:
This merging algorithm seems inefficient. You can create a hashmap for checking features in the ChainConfig contract
```
featuresList := map[string]bool{}
```
Add the following line to the `featureRange` loop
```
featuresList[f.Name] = true 
```
And then you can only add migration features by checking if it exists in the contract
```
for _, feature := range featuresFromState {
	if !featureList[feature.Name] {
		features = append(features, &Feature{
			Name: feature,
			BlockHeight: 0,
			BuildNumber: 0,
			Status: cctypes.Feature_ENABLED,
		})
	}
}
```

Reply:
Actually, time complexity of  both Algorithm is same.
I am creating a universal set and filtering out duplicates using hash map.
Space complexity wise, it uses some extra variables, Ok will trim it.
Do you think we need a utility function which will filter out duplicate features...:)

Reply:
If we don't reuse the utility function and the function is not too long, we better inline it since it make the code easier to read. :)

Reply:
Merge procedure is changed.